ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -937,9 +943,9 @@ protected ByteBuffer filterChunkToRange(ByteBuffer buf) {
         buf.position(0);
         buf.limit(0);
       } else {
-        long startOffsetInThisChunk = chunkIndex == 0 ? resolvedByteRange.getStartOffset() % chunkSize : 0;
+        long startOffsetInThisChunk = chunkIndex == 0 ? resolvedByteRange.getStartOffset() - offset : 0;
         long endOffsetInThisChunk =
-            chunkIndex == (numChunksTotal - 1) ? (resolvedByteRange.getEndOffset() % chunkSize) + 1 : chunkSize;
```

Comment:
One more minor question: is `endOffsetInThisChunk` exclusive or inclusive?
Looks like it is exclusive, right?

Reply:
exclusive since it's used in `Buffer.limit()`.  I can rename it `endOffsetInThisChunkExclusive`

Reply:
If you feel that it is cleaner, you could also express this as `long endOffsetInThisChunkExclusive = Math.min(resolvedByteRange.getEndOffset() - offset + 1, chunkSize)`